### PR TITLE
chore: add linter analysis options

### LINKED ISSUE #3: [chore] - add linting support
Hey, always been lurking in your videos and took a quick look at the codebase. I think the net biggest improvement will be done by just adding lints.

There are a couple of different approaches but new projects use [flutter_lints](https://pub.dev/packages/flutter_lints) which does the job quite well. This would just be a tedius change as there will probably be a lot of small issues that needs to be resolved but should still be very simple and straight forward.

Keep making great videos 😊

### PR DESCRIPTION
**Pull request by BertVanMieghem**
An initial proposal for some linter rules. These are the ones I use personally, they are not very strict but might be a good start.
Check out https://dart.dev/tools/linter-rules

### CODE REVIEW

File: app/exerlog/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    avoid_print: true
+
+    avoid_positional_boolean_parameters: false
+
+    use_build_context_synchronously: false
+
+    lines_longer_than_80_chars: false
+
+    always_put_required_named_parameters_first: false
```
Comment (jorre127) [inline app/exerlog/analysis_options.yaml:33]: I would put this one on as well.

**Comment by jorre127** [comment]
@BertVanMieghem Is this still being worked on ?

**Review by tenshiAMD — CHANGES_REQUESTED** [review]
@BertVanMieghem Hi, please rebase to alpha branch then fix conflicts. Thanks!

**Comment by mfederowicz** [comment]
@BertVanMieghem Hi, what is the status of this PR, maybe some help needed?

**Comment by tenshiAMD** [comment]
> @BertVanMieghem Hi, what is the status of this PR, maybe some help needed?

@mfederowicz sure you can help here

**Comment by mfederowicz** [comment]
Ok @tenshiAMD but what is the plan, I have https://github.com/mfederowicz/exer_log/tree/lintr - cleanedup, 1 ahead to alpha, so should I create another PR?

**Comment by tenshiAMD** [comment]
> Ok @tenshiAMD but what is the plan, I have [https://github.com/mfederowicz/exer_log/tree/lintr](https://github.com/mfederowicz/exer_log/tree/lintr?rgh-link-date=2022-09-12T21%3A14%3A56Z) - cleanedup, 1 ahead to alpha, so should I create another PR?

@mfederowicz yes then we close and reference this PR to the new PR. Thanks.

**Comment by mfederowicz** [comment]
@tenshiAMD ok but can you lookup on content of pubspec.lock, because iam affraid of update some parts, especially material_color_utilities or flutter_riverpod those packages bump up version after:

```
flutter pub get
```

**Comment by mfederowicz** [comment]
ok @tenshiAMD problem with pubspec.loc solved, i pushed changes related only with linter, we can close this PR and focus on: https://github.com/EXERLOG/exer_log/pull/191

**Comment by tenshiAMD** [comment]
> ok @tenshiAMD problem with pubspec.loc solved, i pushed changes related only with linter, we can close this PR and focus on: #191

@mfederowicz ok thanks

@BertVanMieghem we are closing this one for now. thanks.

**Pull request closed without merging**